packages/layout/src/layouts/SwitchableLayout.tsx

```diff
@@ -0,0 +1,79 @@
+import BasicLayout from "./BasicLayout";
+import CollapsibleSidebarLayout from "./CollapsibleSidebarLayout";
+
+interface Properties {
+  children: React.ReactNode;
+  header: React.ReactNode;
+  footer?: React.ReactNode;
+  localSwitcher?: React.ReactNode;
+  menuToggle?: React.ReactNode;
+  mainMenuRoutes: {
+    name: string;
+    route: string;
+    icon?: React.ReactNode;
+  }[];
+  mainMenuOrientation?: "horizontal" | "vertical";
+  userMenu?: React.ReactNode;
+  mainMenu?: React.ReactNode;
+  logoRoute?: string;
+  sidebar?: React.ReactNode;
+  displaySidebarMenuIcon?: boolean;
+  displaySidebar?: boolean;
+  layoutType?: "sidebar" | "basic";
+}
+
+export const SwitchableLayout = ({
+  children,
+  footer,
+  header,
+  sidebar,
+  layoutType = "basic",
+  mainMenuRoutes,
+  displaySidebarMenuIcon,
+  displaySidebar = true,
+  localSwitcher,
+  logoRoute,
+  mainMenu,
+  mainMenuOrientation,
+  menuToggle,
+  userMenu,
+}: Properties) => {
+  return (
```

Comment:
logic with `? : ` is very rigid and doesn't allow for easy extension eg if we want to add another layout type. Could we use methods or `switch...case` that allows better extension?